HIP-0003.md

```diff
@@ -0,0 +1,95 @@
+# HIP-0003 : Authentication using Handshake name
+
+```
+Number:  HIP-0003
+Title:   Authentication using Handshake name
+Type:    Informational
+Status:  Draft
+Authors: Fernando Falci <https://iamfernando/>
+Created: 2020-10-25
+```
+
+## Abstract
+
+This HIP describe how to authenticate a user using his/her Handshake name.
+
+## Motivation
+
+Many websites need to identify the user and proof he/she owns a Handshake name. Many other websites simply need to identify users, which forces them to go into a process to pick a unique username that most of the time can't be consistent across different apps. By using Handshake names as username it allows both: proof ownership of the name and consistent username across unrelated apps.
+
+## Protocol
+
+The basics of this protocol consist of a public and private ECDSA key. The domain exposes the public key via HTTPS request.
+The Service that wants to authenticate the user should provide this user with a unique challenge.
+The User signs the challenge using his/her private key and returns it to the Service, among the public key and the domain name.
+The Service then validates if the challenge is correct and was correct signed. The Service also validates if the domain name indeed exposes that public key.
+
+### Provider
+
+The above-mentioned flow requires an extra piece of software: a user-trusted provider. This provider knows/holds the user's private key. It can run on the client-side, as a browser extension or an installed app; or it can run remotely as a regular web service.
```

Comment:
> The above-mentioned flow requires an extra piece of software: a user-trusted provider. This provider knows/holds the user's private key. It can run client-side, as a browser extension, as an installed app, or run remotely as a regular web service.